codegen/masm/src/emulator/mod.rs

```diff
@@ -1547,6 +1549,19 @@ impl Emulator {
                     return Ok(EmulatorEvent::EnterLoop(body_blk));
                 }
                 Op::Exec(callee) => {
+                    // remove the `::` prefix (absolute path) if any from the
```

Comment:
@bitwalker I've tried to find anything resembling the invocation infra from the VM in the emulator to properly implement absolute paths but did not find any. I put a quick and dirty workaround stripping the `::` prefix.

Reply:
We may need to extend our representation a bit, in order to allow importing MASM modules from source. In particular, we may want to have a variant of `FunctionIdent` called `Callee` used in our MASM representation, like so:

```rust
pub enum Callee {
    MastRoot(RpoDigest),
    Path {
        module: Option<Ident>,
        function: Ident,
        absolute: bool,
    }
}
```

When translating HIR `FunctionIdent` to `Callee`, it would become  a path where the `module` would always be set, and `absolute` would always be `true` (since function identifiers in HIR are always absolute).

When translating MASM `InvocationTarget` to `Callee`, we'd convert `InvocationTarget::AbsolutePath` as an absolute callee, `InvocationTarget::ProcedurePath` as a relative path, `InvocationTarget::ProcedureName` as a relative path, and `InvocationTarget::MastRoot`, as a MAST root. Similar translations could be supported for other `InvocationTarget`-like procedure identifiers, but the MASM opcodes only use `InvocationTarget`.

The emulator then would need to resolve relative paths at runtime, but could treat absolute paths and MAST roots as, well, absolute references. The emulator currently does not track MAST roots at all, so that would be needed to have full compatibility, but would be a pretty simple thing to add.

Reply:
Thank you for the thorough explanation. Do you want me to implement it in this PR? I'd rather make an issue out of it and tackle it in a new PR.

Reply:
@greenhat That's fine by me